### ISSUE TITLE
provide a way to customize the output of the emitter

### ISSUE THREAD

**Issue by softprops** [chyh1990/yaml-rust#26]
currently all yaml strings are being emitted regardless of whether they are actually needed or not. typically you don't need them in yaml. it would be useful to have a way to customize the behavior or to just opt not to quote things by default.

**Comment by chyh1990** [comment]
Yeah, the emitter is not polished yet, I just quote every string to make it safely emit valid YAML document (but may not human friendly).
PR regarding the emitter is welcome.

**Comment by softprops** [comment]
Thanks @chyh1990. Maybe I'll try to see what I I do myself. You're library has been working great so far

**Comment by antoyo** [comment]
I think we can close this issue since pull request #39 has been merged.

**Comment by softprops** [comment]
I think that pull helps with what I wanted out of this issue. There are actually many flavors of string quoting styles defined in the yaml spec that would probably be nice to have in this library. Maybe a separate issue to outline those could be opened in favor of this.

**Comment by radix** [comment]
Would this also have scope for customizing things other than strings? For example there are situations where using single-line `{}` or `[]` constructs for small structures would drastically reduce the size of some of my YAML payloads.

**Comment by hoodie** [comment]
@radix there already is [`emit_node_compact()`](https://github.com/chyh1990/yaml-rust/blob/master/src/emitter.rs#L135), even though this is not public yet. It is currently only used for keys in `Yaml::Hash`. Perhaps you could come up with a nice API to use this one instead.

**Comment by Sytten** [comment]
Just bumping, would be great to have the option to force adding quotes for strings

**Comment by 0xForerunner** [comment]
> Just bumping, would be great to have the option to force adding quotes for strings

bump for that
